File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Game.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "games", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "games_unique", columnNames = {"game_name"})
})
public class Game {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "game_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "location_id")
    private Location location;

    @NotNull
    @Column(name = "date_time", nullable = false)
    private Timestamp dateTime;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "game_status_id")
    private GameStatus gameStatus;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "game_type_id")
    private GameType gameType;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "season_type_id")
    private SeasonType seasonType;

    @NotNull
    @Column(name = "game_name", nullable = false, length = Integer.MAX_VALUE)
    private String gameName;

}

File: umbrella-insurance-react-ui/src/models/games/v1/Game.ts
import { GameType } from "../../gameTypes/v1/GameType";
import { SeasonType } from "../../seasons/seasonTypes/v1/SeasonType";
import { GameStatus } from "../gameStatuses/v1/GameStatus";
import { Location } from "../../geographies/locations/v1/Location";

export class Game {
    id?: number;
    location?: Location;
    dateTime?: string;
    gameStatus?: GameStatus;
    gameType?: GameType;
    seasonType?: SeasonType;
    gameName?: string;

    constructor(obj?: any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.location) {
                this.location = new Location(obj.location);
            }
            if(obj?.dateTime) {
                this.dateTime = obj.dateTime;
            }
            if(obj?.gameStatus) {
                this.gameStatus = new GameStatus(obj.gameStatus);
            }
            if(obj?.gameType) {
                this.gameType = new GameType(obj.gameType);
            }
            if(obj?.seasonType) {
                this.seasonType = new SeasonType(obj.seasonType);
            }
            if(obj?.gameName) {
                this.gameName = obj.gameName;
            }
        }
    }
}